### PR TITLE
chore: trial actor refactor

### PR DESCRIPTION
**Pull request by erikwilson**
## Description

Actor refactor 1/2 for trial [DET-9734]

## Test Plan

N/A - Covered by automated tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[DET-9734]: https://hpe-aiatscale.atlassian.net/browse/DET-9734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by stoksc — APPROVED** [review]
lgtm. there are some things I would do though, just to be safe: run the affected integration tests with `--race` enabled locally since circle does not enable it, run them with coverage enabled, too, and look incase that coverage and "sketchiness" of the code intersect at least (cover the important parts), and, then based on that, maybe circle back around and add another test or two.

File: master/internal/task/allocation_service.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func (as *allocationService) StartAllocation(
 		as.mu.Lock()
 		defer as.mu.Unlock()
 		delete(as.allocations, req.AllocationID)
-		onExit(ref.exited)
+		go onExit(ref.exited)
```
[the remark this answers is not in the document]
Comment (erikwilson) [reply master/internal/task/allocation_service.go]: rearranged locking and left comment

File: master/internal/restore_test.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ func TestDeserExperimentSnapshotIntoCurrent(t *testing.T) {
 				}
 				sm := searcher.NewSearchMethod(single)
 				e.searcher = searcher.NewSearcher(0, sm, expconf.Hyperparameters{})
+				//nolint:govet
```
[the remark this answers is not in the document]
Comment (erikwilson) [reply master/internal/restore_test.go]: updated test to return a reference

**Comment by erikwilson** [comment]
> lgtm. there are some things I would do though, just to be safe: run the affected integration tests with `--race` enabled locally since circle does not enable it, run them with coverage enabled, too, and look incase that coverage and "sketchiness" of the code intersect at least (cover the important parts), and, then based on that, maybe circle back around and add another test or two.

unfortunately it seems the race detector is picking up issues with the mocks for trial integration tests

**Comment by erikwilson** [comment]
> ah, yeah, the mockery mocks are not threadsafe. is it those?

Fixed by implementing String on the actor mock, trial integration tests now pass race detector w/o issue.

**Pull request merged**